modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -652,11 +652,27 @@ def take_2d_labels_or_positional(
 
         if row_labels is not None:
             # Get numpy array of positions of values from `row_labels`
-            row_positions = self.index.get_indexer_for(row_labels)
+            if isinstance(self.index, pandas.MultiIndex):
+                row_positions = np.zeros(len(row_labels), dtype="int64")
+                for idx, label in enumerate(row_labels):
```

Comment:
an in-code comment describing why there's a for-loop instead of a single `.get_locs(labels)` call also would be helpful

Reply:
added